synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -278,30 +271,35 @@ async def action_for_event_by_user(
 
         evaluator = PushRuleEvaluatorForEvent(
             event,
-            len(room_members),
+            room_member_count,
             sender_power_level,
             power_levels,
             relations,
             self._relations_match_enabled,
         )
 
-        # If the event is not a state event check if any users ignore the sender.
-        if not event.is_state():
-            ignorers = await self.store.ignored_by(event.sender)
-        else:
-            ignorers = frozenset()
+        users = list(rules_by_user.keys())
+        profiles = await self.store.get_subset_users_in_room_with_profiles(
+            event.room_id, users
+        )
```

Comment:
I thought the `list` was weird (but fine). It will allocate though.

If you're a microoptimising speed freak you could pass the dict directly. Or pass `dict.keys()` directly. Change the store function to accept a Collection[str] (which is what `make_in_list_sql_clause` actually wants).